src/common.cc

```diff
@@ -394,6 +430,31 @@ namespace sharp {
         image.get_image()->Type = image.guess_interpretation();
         image = image.cast(VIPS_FORMAT_UCHAR);
         imageType = ImageType::RAW;
+      } else if (descriptor->textValue.length() > 0) {
+        // Create a new image with text
+        vips::VOption *textOptions = VImage::option()
+          ->set("align", descriptor->textAlign)
+          ->set("justify", descriptor->textJustify)
+          ->set("rgba", descriptor->textRgba)
+          ->set("spacing", descriptor->textSpacing);
+        if (descriptor->textWidth > 0) {
+          textOptions->set("width", descriptor->textWidth);
+        }
+        // Ignore dpi if height is set
+        if (descriptor->textWidth > 0 && descriptor->textHeight > 0) {
+          textOptions->set("height", descriptor->textHeight);
+        } else if (descriptor->textDpi > 0) {
+          textOptions->set("dpi", descriptor->textDpi);
+        }
+        if (descriptor->textFont.length() > 0) {
+          textOptions->set("font", const_cast<char*>(descriptor->textFont.data()));
+        }
+        if (descriptor->textFontfile.length() > 0) {
+          textOptions->set("fontfile", const_cast<char*>(descriptor->textFontfile.data()));
+        }
+        image = VImage::new_memory().text(const_cast<char *>(descriptor->textValue.data()), textOptions);
```

Comment:
I think you can just call `VImage::text` here.
```suggestion
        image = VImage::text(const_cast<char *>(descriptor->textValue.data()), textOptions);
```

Reply:
Done